packages/x-tree-view/src/internals/plugins/useTreeViewNodes/useTreeViewNodes.ts

```diff
@@ -154,7 +168,16 @@ export const useTreeViewNodes: TreeViewPlugin<UseTreeViewNodesSignature> = ({
     return childrenIds;
   };
 
+  const isItemUpdatedPreventedRef = React.useRef(false);
+  const preventItemUpdate = useEventCallback(() => {
```

Comment:
This PR makes the item register earlier in `SimpleTreeView`, which is great but causes the empty `props.items` array to wipe the state after the initialization.
The simplest solution I found is just to have `useTreeViewJSXNodes` prevents any state update based on `props.items` with this method.